module/VuFind/src/VuFind/ILS/Driver/Alma.php

```diff
@@ -865,66 +868,67 @@ public function getMyFines($patron)
      */
     public function getMyHolds($patron)
     {
-        $xml = $this->makeRequest(
-            '/users/' . $patron['id'] . '/requests',
-            ['request_type' => 'HOLD']
-        );
         $holdList = [];
-        foreach ($xml as $request) {
-            $lastInterestDate = $request->last_interest_date
-                ? $this->dateConverter->convertToDisplayDate(
-                    'Y-m-dT', (string)$request->last_interest_date
-                ) : null;
-            $available = (string)$request->request_status === 'On Hold Shelf';
-            $lastPickupDate = null;
-            if ($available) {
-                $lastPickupDate = $request->expiry_date
+        $offset = 0;
+        $totalCount = 1;
+        $allowCancelingAvailableRequests
+            = !isset($this->config['Holds']['allowCancelingAvailableRequests'])
+                || $this->config['Holds']['allowCancelingAvailableRequests'];
+        while ($offset < $totalCount) {
+            $xml = $this->makeRequest(
+                '/users/' . $patron['id'] . '/requests',
+                ['request_type' => 'HOLD', 'offset' => $offset, 'limit' => 100]
+            );
+            $offset += 100;
+            $totalCount = (int)$xml->attributes()->{'total_record_count'};
+            foreach ($xml as $request) {
+                $lastInterestDate = $request->last_interest_date
                     ? $this->dateConverter->convertToDisplayDate(
-                        'Y-m-dT', (string)$request->expiry_date
+                        'Y-m-dT',
+                        (string)$request->last_interest_date
                     ) : null;
+                $available = (string)$request->request_status === 'On Hold Shelf';
+                $lastPickupDate = null;
+                if ($available) {
+                    $lastPickupDate = $request->expiry_date
+                        ? $this->dateConverter->convertToDisplayDate(
+                            'Y-m-dT',
+                            (string)$request->expiry_date
+                        ) : null;
+                    $lastInterestDate = null;
+                }
+                $requestStatus = (string)$request->request_status;
+                if (!$available || $allowCancelingAvailableRequests) {
```

Comment:
`$updateDetails = (!$available || $allowCancelingAvailableRequests) ? (string)$request->request_id : '';` would be more concise here. Since this is a very long function, every little bit helps. :-)